prog/probe.go

```diff
@@ -221,3 +233,28 @@ func getFirstAddressOf(name string) (string, error) {
 		return "", fmt.Errorf("No address found for %s", name)
 	}
 }
+
+// TODO(paulbellamy): where should this live?
+func pluginsReporter(pluginRegistry *plugins.Registry) probe.Reporter {
+	return probe.ReporterFunc("plugins", func() (report.Report, error) {
+		rpt := report.MakeReport()
+		pluginRegistry.Implementors("reporter", func(plugin *plugins.Plugin) {
+			pluginReportJSON := map[string]interface{}{}
+			if err := plugin.Call("Report", nil, &pluginReportJSON); err != nil {
```

Comment:
You need some deadline on this, don't want a plugin to block the whole reporting cycle.